api/src/main/java/org/openmrs/api/context/OpenmrsApplicationContextConfig.java

```diff
@@ -0,0 +1,208 @@
+import org.openmrs.api.ProviderService;
+import org.openmrs.api.SerializationService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.VisitService;
+import org.openmrs.api.db.ContextDAO;
+import org.openmrs.api.db.hibernate.AuditableInterceptor;
+import org.openmrs.api.db.hibernate.DbSessionFactory;
+import org.openmrs.api.db.hibernate.HibernateSessionFactoryBean;
+import org.openmrs.api.impl.AdministrationServiceImpl;
+import org.openmrs.api.impl.GlobalLocaleList;
+import org.openmrs.api.impl.OrderServiceImpl;
+import org.openmrs.api.impl.PersonNameGlobalPropertyListener;
+import org.openmrs.hl7.HL7Service;
+import org.openmrs.logging.LoggingConfigurationGlobalPropertyListener;
+import org.openmrs.messagesource.MessageSourceService;
+import org.openmrs.notification.AlertService;
+import org.openmrs.notification.MessageService;
+import org.openmrs.scheduler.SchedulerService;
+import org.openmrs.util.ConfigUtil;
+import org.openmrs.util.HttpClient;
+import org.openmrs.util.LocaleUtility;
+import org.openmrs.util.LocationUtility;
+import org.openmrs.util.TestTypeFilter;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+
+@Configuration
+@ComponentScan(
+    basePackages = "org.openmrs",
+    includeFilters = {
+        @ComponentScan.Filter(type = FilterType.ANNOTATION, classes = Handler.class),
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = OpenmrsProfileIncludeFilter.class)
+    },
+    excludeFilters = {
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = TestTypeFilter.class),
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = OpenmrsProfileExcludeFilter.class)
+    }
+)
+public class OpenmrsApplicationContextConfig {
+
+    @Bean
+    public EventListeners clearOpenmrsEventListeners() {
+        EventListeners eventListeners = new EventListeners();
+        eventListeners.setGlobalPropertyListenersToEmpty(false);
+        return eventListeners;
+    }
+
+    @Bean
+    public LocaleUtility localeUtility() {
+        return new LocaleUtility();
+    }
+
+    @Bean
+    public LocationUtility locationUtility() {
+        return new LocationUtility();
+    }
+
+    @Bean
+    public ConfigUtil configUtilGlobalPropertyListener() {
+        return new ConfigUtil();
+    }
+
+    @Bean
+    public PersonNameGlobalPropertyListener personNameGlobalPropertyListener() {
+        return new PersonNameGlobalPropertyListener();
+    }
+
+    @Bean
+    public LoggingConfigurationGlobalPropertyListener loggingConfigurationGlobalPropertyListener() {
+        return new LoggingConfigurationGlobalPropertyListener();
+    }
```

Comment:
For simple beans, please just annotate classes with `@Component` instead of adding them to the config. It makes sense to add classes to the config if we want to specifically configure something or cannot annotate classes as they come from external libraries. Also if there was an id used in xml, please set the same id using annotation.